django/contrib/admin/checks.py

```diff
@@ -88,17 +103,16 @@ def check_dependencies(**kwargs):
                 "the admin application.",
                 id='admin.E404',
             ))
-    if ('django.contrib.auth.middleware.AuthenticationMiddleware'
-            not in settings.MIDDLEWARE):
+
+    if not _contains_subclass('django.contrib.auth.middleware.AuthenticationMiddleware', settings.MIDDLEWARE):
         errors.append(checks.Error(
-            "'django.contrib.auth.middleware.AuthenticationMiddleware' must "
+            "'django.contrib.auth.middleware.AuthenticationMiddleware' (or a subclass) must "
```

Comment:
The messages corresponding to these will need updating in `docs/ref/checks.txt`.